### PR TITLE
feat: Update assigned state and fix for wrong notifications

### LINKED ISSUE #8999: Update assigned state
Whenever an my company is assigned, we'd need to provide an update in the UI. We need to introduce this to the state and have a dedicated method in the facade. 

For example: budget assigned to cost-center

Wait for #8976 to be done so we have a strategy, then tackle this one

We like to notification message to contain more context (the item.name for example)
double check: after assigning, double check if the user is actually assigned.

### LINKED ISSUE #9224: Wrong notification while toggling items
After enable item I can see notification `"<item> disabled successfully"` and while disable: `"<item> enabled successfully"`. It should be vice versa

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #8999, #9224

### CODE REVIEW

File: feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts [hunk 1]
```diff
@@ -94,7 +95,17 @@ export class ToggleStatusComponent<T extends BaseItem> implements OnDestroy {
   }
 
   protected update(item: T): void {
-    this.itemService.update(item[this.key], this.getPatchedItem(item));
+    this.itemService
+      .update(item[this.key], this.getPatchedItem(item))
+      .pipe(
+        take(1),
+        tap((data) => {
```
Comment (Pucek9) [inline feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts]: do we want to display here any error message for FAIL?

File: feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts [hunk 2]
```diff
@@ -120,14 +131,6 @@ export class ToggleStatusComponent<T extends BaseItem> implements OnDestroy {
     this.itemActiveState = { ...item };
```
Comment (tobi-or-not-tobi) [inline feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts]: The `itemActiveState` is no longer needed, as it was used before to verify if the item was activated. we should drop this line as well as the `itemActiveState` property.

File: feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts]: yes, that would be nice, but I assume most errors might end up in the current global message component. If that's the case, it's fine for now.

File: feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts [hunk 3]
```diff
@@ -94,7 +95,17 @@ export class ToggleStatusComponent<T extends BaseItem> implements OnDestroy {
   }
 
   protected update(item: T): void {
-    this.itemService.update(item[this.key], this.getPatchedItem(item));
+    this.itemService
+      .update(item[this.key], this.getPatchedItem(item))
+      .pipe(
+        take(1),
```
Comment (tobi-or-not-tobi) [inline feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts:99]: What happens if you move away from this component, i.e. if there's a delay? You'd go in the notify method and quit likely have an error. Probably savest to clean up subscription anyway.

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts (hunk 1 above)
Comment (Pucek9) [reply feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts]: Yep, but we don't have a list of possible backend errors / codes so we don't have that in translations. E.g `common:httpHandlers.validationErrors.invalid.budgetData`

File: feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts (hunk 3 above)
Comment (Pucek9) [reply feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts:99]: I tried with slow 3g mode, disable/enable and go to other page. In if we do that fast, will not see notification and no error, else we can see notification and then we are redirected.

File: feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts (hunk 2 above)
Comment (Pucek9) [reply feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts]: cleaned

File: feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts (hunk 3 above)
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts:99]: I'd probably add an explicit rxjs `delay` of 5000 and see if you face an error or leak.

File: feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/components/shared/organization-detail/toggle-status-action/toggle-status.component.ts]: As discussed we move the conversations regarding errors to new ticket. 

I'd nit about changing this code to: 

```ts
this.itemService
      .update(item[this.key], this.getPatchedItem(item))
      .pipe(
        take(1),
        filter((data) => data.status === LoadStatus.SUCCESS)
      )
      .subscribe((data) => this.notify(data.item));
```

**Review by tobi-or-not-tobi — DISMISSED** [review]
Good stuff, one small optimization you could think of, but without it i'm happy too.

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**